src/lib/support/ScopedBuffer.h

```diff
@@ -142,7 +142,7 @@ class ScopedMemoryBuffer : public Impl::ScopedMemoryBufferBase<MemoryManagement>
     inline T * Get() { return static_cast<T *>(Base::Ptr()); }
     inline T & operator[](size_t index) { return Get()[index]; }
 
-    inline const T * Get() const { return static_cast<T *>(Base::Ptr()); }
+    inline const T * Get() const { return static_cast<const T *>(Base::Ptr()); }
```

Comment:
Why this change?  Returning `T* ` from a method that returns `const T*` is fine, and casting both type and const-ness together is not great, imo....

Reply:
Previous was not const correct: Base::Ptr() returns a const because the method is const. so static_cast to non_const was failing because it was removing constness (even though it was added right back as a return type).